share/ingest/change_builder.py

```diff
@@ -0,0 +1,255 @@
+import logging
+import pendulum
+import datetime
+
+from django.apps import apps
+from django.conf import settings
+from django.contrib.contenttypes.models import ContentType
+
+from share.disambiguation import GraphDisambiguator
+from share.models import Change, ChangeSet
+from share.util import IDObfuscator
+
+
+logger = logging.getLogger(__name__)
+
+
+class ChangeBuilder:
+
+    @classmethod
+    def build_change_set(cls, graph, normalized_datum, instance_map=None, disambiguate=False):
+        assert not (instance_map and disambiguate), 'Either provide instance_map or disambiguate=True, not both'
+
+        user = normalized_datum.source  # "source" here is a ShareUser
+        source = user.source
+
+        if disambiguate:
+            instance_map = GraphDisambiguator(source).find_instances(graph)
+        elif instance_map is None:
+            instance_map = {}
+
+        differs = [cls(n, source, instance_map) for n in graph.topologically_sorted()]
```

Comment:
Maybe `differences`?

Reply:
Oh whoops, ChangeBuilder used to be called NodeDiffer, because that felt like a good idea at the time. Should probably name this `builders` or something like that.